src/_pytest/nodes.py

```diff
@@ -159,6 +159,13 @@ class Node(metaclass=NodeMeta):
     Collector subclasses have children; Items are leaf nodes.
     """
 
+    # Implemented in the legacypath plugin.
+    #: A ``LEGACY_PATH`` copy of the :attr:`path` attribute. Intended for usage
+    #: for methods not migrated to ``pathlib.Path`` yet, such as
+    #: :meth:`Item.reportinfo`. Will be deprecated in a future release, prefer
+    #: using :attr:`path` instead.
+    fspath: LEGACY_PATH
+
```

Comment:
@bluetech What's your take on this? From what I understand, the aim of #9208 was to decouple legacy stuff like `fspath`, so this might seem out of place. However, I'd really prefer to keep this attribute documented (even more so because it's not deprecated yet!), and having this here is the only way I found to make Sphinx pick it up.

Reply:
Two aspects to this - docs and typing. For the docs it might work to add it in reference.rst as regular sphinx attr instead of autodoc? As for typing, without this PR it would show up as a missing attribute error, but maybe it's a good thing?

Anyway, I'm fine with whatever you think is best (including this approach).

Reply:
I feel like having this as a "missing" attribute would be wrong - especially given that we explicitly un-deprecated it. Until we deprecate (or, ideally, remove) it, I feel like typing should know that it's there too.